core/modules/node/node.entity.inc

```diff
@@ -800,6 +800,13 @@ class NodeStorageController extends EntityDatabaseStorageController {
         $build['#contextual_links']['node'] = array('node', array($node->nid));
       }
 
+      // Show a warning if the content type is configured to have hidden path.
```

Comment:
we just need to add....
`if user_access('view hidden paths') {`
around the whole set of new code.

Reply:
That's not all. The whole bit needs to move as I mentioned in my comment. Should be in `node_page_view`, not here.